cmd/maincmd/main.go

```diff
@@ -90,22 +104,48 @@ func Main() {
 		Handler: h.Handler(),
 	}
 
+	// shutdownCh triggers graceful shutdown on SIGINT or SIGTERM
+	shutdownCh := make(chan os.Signal, 1)
+	signal.Notify(shutdownCh, syscall.SIGINT, syscall.SIGTERM)
+
+	// exitCh will be closed when it is safe to exit, after graceful shutdown
+	exitCh := make(chan struct{})
+
+	go func() {
+		sig := <-shutdownCh
+		serverLog("shutdown started by signal: %s", sig)
+
+		shutdownTimeout := maxDuration + 1*time.Second
+		ctx, cancel := context.WithTimeout(context.Background(), shutdownTimeout)
+		defer cancel()
+
+		server.SetKeepAlivesEnabled(false)
```

Comment:
@mccutchen I noticed you didn't do this with SSO, was there a reason for that?

Reply:
@Integralist good eye! I had it in my original SSO PR (that one followed this one), but Justin rightly [asked me to explain it](https://github.com/buzzfeed/sso/pull/262#discussion_r340689367).  To save you a click, here's my response:

> I can't [explain it]!  I _thought_ I had copied this from [the stdlib's Shutdown example](https://golang.org/pkg/net/http/#Server.Shutdown), but looking at it again I do not see this called.
> 
> Went ahead and dropped this.

Just never followed up here.